### DISCUSSION TITLE
One rauc-bundle recipe for multiple generated rauc-bundles

### DISCUSSION THREAD

**Discussion by Bouchellal** [rauc/meta-rauc#346] [Q&A] [answered]
Hello,

I am using rauc-bundle for my system, i have a linux system and 2 subdevices, so my recipe looks like this:

```
DESCRIPTION = "RAUC bundle generator"

LICENSE = "MIT"
LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"

inherit bundle
inherit dm-verity-img

RAUC_BUNDLE_COMPATIBLE = "${MACHINE}"
RAUC_BUNDLE_VERSION = "${DISTRO_VERSION}"
RAUC_BUNDLE_DESCRIPTION = "Rauc update Bundle"
RAUC_BUNDLE_SLOTS = **"rootfs subdevice1 subdevice2"**
RAUC_BUNDLE_FORMAT = "verity"
RAUC_SLOT_rootfs = "production-image"
RAUC_SLOT_rootfs[fstype] = "ext4.verity"

SRC_URI += " file://hook.sh"
RAUC_BUNDLE_HOOKS[file] = "hook.sh"
RAUC_SLOT_rootfs[hooks] = "pre-install;post-install"

RAUC_SLOT_subdevice1 = "subdevice1-image"
RAUC_SLOT_subdevice1[file] = "subdevice1.bin"
RAUC_SLOT_subdevice1[hooks] = "install"

RAUC_SLOT_subdevice2 = "subdevice2-firmware"
RAUC_SLOT_subdevice2[file] = "subdevice2"
RAUC_SLOT_subdevice2[hooks] = "install"
```

I am successfully generating a rauc-bundle with the three images for each slot, now i want to generate another rauc-bundle for each slot individually. should i write an entire recipe for each one, or there is a trick for this purpose ?

**Comment by ejoerns** [comment]
You could for example overwrite `RAUC_BUNDLE_SLOTS` from your `local.conf` and remove/include only the slot images that should appear in the final bundle.

**Reply by Bouchellal** [reply]
I mean, i want the recipe to build first bundle with three slots, then build another bundle with only one specific slot. All this, during one "bitbake rauc-bundle" cmd. Is this use case covered ? if no, do you have an idea how to implement this using bitbake task trick 
?

**Reply by ejoerns** [reply, marked as answer]
You can have multiple recipes that share the basic information in a common `.inc` and set `RAUC_BUNDLE_SLOTS` individually. With bitbake you can then just do `bitbake bundle-1 bundle-2 bundle-3` to generate all in one run.

**Reply by Bouchellal** [reply]
that's a great suggestion, thank you !
